### ISSUE TITLE
Cameras not showing in 0.20.1

### ISSUE THREAD

**Issue by durosity** [Colorado4Wheeler/HomeKit-Bridge#89]
I previously reported in another thread that cameras disappeared when upgrading plugin, but then I could re-add them and it worked.  I also had an issue related where it disappeared and wouldn’t re-add, but changing the device MAC address allowed Home to re-add it.  Alas with v0.20.1 this has happened again but changing the address doesn’t help - the “server” device appears but the camera devices do not.  I have not yet changed anything on this setup (Other than MAC address) in case there was anything you wanted me to check - any suggestions before I try the usual troubleshooting?

**Comment by Colorado4Wheeler** [comment]
FFS, I can't catch a break here.  Again, down the rabbit hole.  What MAC address are you changing?  There's no reason this update should have adversely impacted your existing camera feeds.  Please dump your config on the camera server so we can see what's different and/or wrong.

**Comment by durosity** [comment]
Sorry.. I was tempted not to post it for a while.. just felt that’d be wrong!  This is the MAC address under advanced settings for HB User.  I’ll PM you the config file in indigo forum as contains passwords.

**Comment by Colorado4Wheeler** [comment]
Ok, that's a username and if you have to change that to get things to come back then that wasn't the plugin but rather HomeKit itself that went sideways.  Aside from **me** looking at your config, what do **you** see that may be different or wrong?  Can you put that URL into a browser and pull the feed?

I'm also running SecuritySpy and obviously my feeds are working (or I wouldn't have released this).

**Comment by Colorado4Wheeler** [comment]
FYI, you know the config and log dump is in the Advanced Plugin Actions right?  I just don't want you to have to poke around the folders.

**Comment by durosity** [comment]
Yes the video sources work fine when loaded directly via safari.  

I don’t think the username is making any difference in this case, I just posted that info as a FYI in case it was related.  I’ve just tried setting up a separate Home on my wife iPhone which is under a separate iCloud ID and it won’t show up in there, but a brand new device with a camera added in HKB does show available to add on both accounts.

**Comment by durosity** [comment]
(I’m fairly sure as you said before it’s likely my home configuration that’s causing the issue.  I can easily resolve it by creating a new device in indigo with the devices.. I just didn’t want to do it in case it was something you wanted to investigate)

**Comment by Colorado4Wheeler** [comment]
It sounds like your HomeKit got messed up, I don't know why that would be.  I haven't lost my cameras yet, although I've had some HomeKit resets.  

> (I’m fairly sure as you said before it’s likely my home configuration that’s causing the issue. I can easily resolve it by creating a new device in indigo with the devices.. I just didn’t want to do it in case it was something you wanted to investigate)

I don't know about that, I've had some HomeKit reset issues myself and it MAY be related to cameras but the jury is still out.  My cameras haven't ever disappeared though.  The fact that they show up on your wife's phone indicates that it's a HomeKit issue.

Are your cameras with other devices or on a server by themselves?

**Comment by Colorado4Wheeler** [comment]
By the way, you could simply try removing the accessory from HomeKit that has your cameras and see if it appears on the list again.  You would have to re-assign rooms but that's not a big deal for a handful of cameras.  Now for 300 devices it is - I know because i've had to do it a half dozen times in the past couple weeks.  😡

**Comment by durosity** [comment]
Just to clarify its the NEW server I created that shows up both on my iPhone and my wifes, and the existing one does not on either.  And it doesn’t show up on my wife’s home account that’s shared from me either.  

And yes they’re in a server all by themselves.  It seemed sensible to split up my installation based on type so I have one for alarm, cameras, sensors, etc.  

I don’t have the ability to remove this from Home.. they simply disappear with no trace.

**Comment by Colorado4Wheeler** [comment]
Yes, I rogered that, but if a **new** server shows up then that means the **old** server is unresponsive in HomeKit, it's not a plugin problem otherwise the new server would be toast too.

> I don’t have the ability to remove this from Home.. they simply disappear with no trace.

Wow, really?  That's super odd.  If the plugin stopped serving or puked entirely then you would see the last image that was downloaded before the blow up, they wouldn't just go away entirely.  This may sound stupid, but are you sure they aren't still there but in a different room now?  I've lost rooms multiple times and if your cameras are hanging around in the default room and not favorite then you may think they are gone.

**Comment by durosity** [comment]
Yeah that’s what I figured, that KH had lost track of it.. not the plugin!

And yeah definitely not in default room.  I keep a very close eye on that one.  I’ve not yet had a situation where HK has lost rooms for me.  If I modify a device it might drop from that room as HK recognises it as a new device, but that’s not unexpected.

**Comment by Colorado4Wheeler** [comment]
Dropping entirely is such a strange issue.  I've had my entire HomeKit disappear on me a few times over the past week, once it didn't come back and twice it did but it was panic as I wondered where my entire setup went.  It was odd, too, they all disappeared and then 30 minutes later they all showed up.

Have you browsed every room to see if even the server they were sitting on is still in HomeKit that you could try to delete it?  And you said you changed the username to no avail, which is also strange because that should force HK to see it again - although I've had server names get "banned" from HK where I had to change the server name to get things working (just happened last weekend actually).

**Comment by Colorado4Wheeler** [comment]
Something I'm trying, because my rooms keep going away, is adding a device to my camera server just to make sure Homebridge isn't encountering some error because there are no devices.  In the next update or so I'm going to put some logic in to not initialize the Homebridge-Indigo2 platform if there are no devices but I haven't gotten to that yet because I sit around all day with my thumb in my ear and do absolutely nothing 🙄.

**Comment by durosity** [comment]
Yep checked every room!  This isn’t the first time something has just disappeared from Home.. even back when it was just Homebridge-Indigo and HBB.  I’ve read of other people having similar issues before too.. particularly the Hue bridge back a few years ago.  

Well it does seem that way.. you’re clearly just lazy.. you have the same motto as me “Hard work may pay off after a time, but laziness always pays off now” hehe!

**Comment by Colorado4Wheeler** [comment]
> Hard work may pay off after a time, but laziness always pays off now

LMAO.

**Comment by durosity** [comment]
Aww you found that funny.. you should know by now my sense of humour is awful.  You’re clearly defective.

Here’s a banana.  
![image](https://user-images.githubusercontent.com/17961550/38061323-b703a056-32e6-11e8-871a-f90912d098b4.png)

**Comment by Colorado4Wheeler** [comment]
Were you able to get your cameras back?

**Comment by durosity** [comment]
Not using the same HBK server, I had to delete that one and setup a new one.  Is there anything you’d like me to test/log/etc before upgrading to beta 21?

**Comment by Colorado4Wheeler** [comment]
There’s not really much I can tell you to do because it sounds like a HomeKit issue rather than a plugin issue.  You could try rebooting your Indigo server in case there’s a rogue Homebridge process running but I would be surprised if that were the case.  One way you could test would be to disable the plugin and do a ‘ps ax | grep homebridge’ in terminal to see if anything is still running.  Instead of deleting next time try just changing the name and user, that should be enough to prime the HomeKit pump, you shouldn’t have to delete.  If you DO delete, then create a new server and use the move utility to move your devices from one to the other, at least you can minimize the work.

**Comment by durosity** [comment]
Fascinating.  Nothing changed, no scenes or cameras disappeared!  One thing I did do differently to usual was I had the home app open when I upgraded the server.  That may be relevant.

**Comment by Colorado4Wheeler** [comment]
I’m not sure it is.  I think the new change I made to the plugin may be the reason why, I believe that you may have problems because there were no devices being sent to Homebridge with your cameras, so when the HB-Indigo2 tried to initialize it was getting “0 accessories” and that caused problems.  I added code to say that if no accessories accompany your cameras then don’t add the platform for HB-Indigo2 and only add the platform for cameras.

This has been a working theory between @webdeck and I all week.  

This also rides on the tail of another issue I had been having of all of my rooms and automation getting reset constantly, but once I added that code I haven’t had a reset yet.  I suspect you and I may have been some of the only ones who were running cameras on servers by themselves with no other accessories on the server and that may be why we had more HomeKit problems.

This is all hypothetical, the jury is still out on this, but I suspect that this may be why you didn’t have a reset.  Coincidentally, while doing some coding today I totally lost my server and all accessories when I changed something that resulted in Homebridge seeing no accessories - however, once I fixed the problem they all came back so not quite the same as yours but I also have to wonder if your cameras were all still gone and you updated to this release if they would have returned.

**Comment by durosity** [comment]
Doh, unfortunately it happened again!  Changing a spelling mistake on a motion sensor caused all cameras and scenes to disappear. (completely separate server device to cameras and no scene used any devices on that server either)

**Comment by Colorado4Wheeler** [comment]
I think that is coincidental, considering I’ve had that happened a few times myself under entirely different circumstances, I doubt it had to do with changing names.  For me, since doing the camera change, mine has stayed intact so far for several days.  

I think you and I are the only ones experiencing this particular issue.  I wonder if there’s a common thread.
